python/res/fm/util/template_render.py

```diff
@@ -0,0 +1,79 @@
+gen_kw_export_file = "parameters";
+
+import jinja2
+import json
+import os
+import yaml
+
+#default export file name found in enkf_defaults.hpp
+DEFAULT_GEN_KW_EXPORT_NAME = "parameters.json"
+
+class TemplateRender:
```

Comment:
No reason to keep this in a class.. I suggest to remove the class and move the functions out of the `util` submodule and into a `templating` submodule. Then the submodule provides the same "namespacing" as the class did earlier, but in a more pythonic way ;)

Reply:
Whether it is implemented in a class or not is not that important to me; but I would would like the *load dictionary from `parameters.json`* functionality to be in separate, testable+++ unit. That way we can fasilitate a two level approach to creating input files from the `parameters.json` file:

1. The simplest will be plain template instantiation - as provided by this PR.

2. The second will be the ability to create files based on the content of `parameters.json` in an arbitrary way[1]; then people can create forward model executable like this:

```
#!/usr/bin/env python
from res.util import parameters   

value1 = parameters["MAJOR"]["val1"]
value2 = parameters["MAJOR"]["val2"]

if value1 > value2:
    alternative1()
else:
    alternative2()
```
The important point is that we a simple way to get the `parameters.json` loaded into a dictionary structure; from there on people write their own script.



[1]: Yes - I know the jinja2 templates are turing complete - that does not necessary imply that they will always be the most convenient way to make your files.

Reply:
> [1]: Yes - I know the jinja2 templates are turing complete - that does not necessary imply that they will always be the most convenient way to make your files.

In the same way as most programming languages being turing complete, and that we still have numerous of them. So no need for a discussion here, you are definitively right ;)

Reply:
> The important point is that we a simple way to get the parameters.json loaded into a dictionary structure; from there on people write their own script.

I agree, but I think we should only make one promise here. Either the API is that we dump the parameters as _json_ to `parameters.json`, which is quite easy to load in all programming languages. Or we say that we can give you the parameters in a data structure in _Python_ and you can use that. Which is slightly easier in _Python_, but less general..? Also, I would in the later case hope that we then expose a function `load_parameters` that they can call?

Reply:
> Or we say that we can give you the parameters in a data structure in Python and you can use that. Which is slightly easier in Python, but less general..? Also, I would in the later case hope that we then expose a function load_parameters that they can call?

Hehe - so much for clear communication; I have argued for the second alternative you mention - including to expose well defined functionality to load parameters; to ensure that the `load_parameters()` functionality is published in some way - be it a a class (method) or a function - has been the whole purpose of my commenting in this thread.

Reply:
Hehe! That is fine by me :) Just not in a class please.. Let us save the dummy classes with no state for the _Java_-developers.

Reply:
> with no state for the Java-developers.

That hypothetical class would have state - the name of the file in question; i.e. `parameters.json`.

Reply:
Yeah, but that is immutable right? So might as well be an member of the submodule? Anyway, I think we agree on this, so lets just see how it turns out :+1:

Reply:
>Yeah, but that is immutable right? 

yes - it is immutable; if we fetch it from the compiled code (which we ideally should) - there will be some fencing involved.

> So might as well be an member of the submodule? 

Yes

> Anyway, I think we agree on this, so lets just see how it turns out +1

Yes +1